src/GafferScene/MotionPath.cpp

```diff
@@ -0,0 +1,443 @@
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferScene/MotionPath.h"
+
+#include "GafferScene/Isolate.h"
+#include "GafferScene/SceneAlgo.h"
+
+#include "Gaffer/Context.h"
+#include "Gaffer/NumericPlug.h"
+
+#include "IECoreScene/CurvesPrimitive.h"
+
+#include "OpenEXR/ImathMatrixAlgo.h"
+
+using namespace IECore;
+using namespace IECoreScene;
+using namespace Gaffer;
+using namespace GafferScene;
+
+namespace
+{
+
+static InternedString g_lightsSetName( "__lights" );
+static InternedString g_defaultLightsSetName( "defaultLights" );
+static InternedString g_camerasSetName( "__cameras" );
+
+} // namespace
+
+GAFFER_NODE_DEFINE_TYPE( MotionPath );
+
+size_t MotionPath::g_firstPlugIndex = 0;
+
+MotionPath::MotionPath( const std::string &name )
+	:	FilteredSceneProcessor( name, PathMatcher::NoMatch )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+
+	ValuePlugPtr startPlug = new ValuePlug( "start", Plug::In );
+	startPlug->addChild( new IntPlug( "mode", Plug::In, (int)FrameMode::Relative, /* min */ (int)FrameMode::Relative, /* max */ (int)FrameMode::Absolute ) );
+	startPlug->addChild( new FloatPlug( "frame", Plug::In, -2 ) );
+	addChild( startPlug );
+
+	ValuePlugPtr endPlug = new ValuePlug( "end", Plug::In );
+	endPlug->addChild( new IntPlug( "mode", Plug::In, (int)FrameMode::Relative, /* min */ (int)FrameMode::Relative, /* max */ (int)FrameMode::Absolute ) );
+	endPlug->addChild( new FloatPlug( "frame", Plug::In, 2 ) );
+	addChild( endPlug );
+
+	addChild( new IntPlug( "samplingMode", Plug::In, (int)SamplingMode::Variable, /* min */ (int)SamplingMode::Variable, /* max */ (int)SamplingMode::Fixed ) );
+	addChild( new FloatPlug( "step", Plug::In, 1, 1e-6 ) );
+	addChild( new IntPlug( "samples", Plug::In, 2, 2 ) );
```

Comment:
Sorry, just one more thing. Is the default of `2` here a bit miserly? It doesn't even give one sample per frame for the default start/end. Can we make it `10` or something?

Reply:
Addressed in 8147814a2a77e289bafad77117ec260dd912ca15